app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,458 @@
+            setChannelLoading: PropTypes.func.isRequired
+        }).isRequired,
+        channelId: PropTypes.string,
+        channelName: PropTypes.string,
+        channelTeamId: PropTypes.string,
+        currentTeamId: PropTypes.string.isRequired,
+        currentUserId: PropTypes.string.isRequired,
+        focusedPostId: PropTypes.string.isRequired,
+        isPermalink: PropTypes.bool,
+        myMembers: PropTypes.object.isRequired,
+        navigator: PropTypes.object,
+        onClose: PropTypes.func,
+        onPermalinkPress: PropTypes.func,
+        onPress: PropTypes.func,
+        postIds: PropTypes.array,
+        theme: PropTypes.object.isRequired
+    };
+
+    static defaultProps = {
+        onPress: () => true,
+        postIds: []
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired
+    };
+
+    constructor(props, context) {
+        super(props);
+
+        const {postIds, channelName} = props;
+        let show = false;
+
+        if (postIds && postIds.length >= 10) {
+            show = true;
+        }
+
+        if (!show) {
+            this.loadPosts(props, context);
+        }
+
+        this.state = {
+            title: channelName,
+            show,
+            error: '',
+            retry: false
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channelName !== nextProps.channelName) {
+            this.setState({title: nextProps.channelName});
+        }
+
+        if (this.props.focusedPostId !== nextProps.focusedPostId) {
+            this.setState({show: false}, () => {
+                if (nextProps.postIds && nextProps.postIds.length < 10) {
+                    this.loadPosts(nextProps);
+                } else {
+                    this.setState({show: true});
+                }
+            });
+        }
+    }
+
+    goToThread = wrapWithPreventDoubleTap((post) => {
+        const {actions, navigator, theme} = this.props;
+        const channelId = post.channel_id;
+        const rootId = (post.root_id || post.id);
+
+        actions.loadThreadIfNecessary(rootId, channelId);
+        actions.selectPost(rootId);
+
+        const options = {
+            screen: 'Thread',
+            animated: true,
+            backButtonTitle: '',
+            navigatorStyle: {
+                navBarTextColor: theme.sidebarHeaderTextColor,
+                navBarBackgroundColor: theme.sidebarHeaderBg,
+                navBarButtonColor: theme.sidebarHeaderTextColor,
+                screenBackgroundColor: theme.centerChannelBg
+            },
+            passProps: {
+                channelId,
+                rootId
+            }
+        };
+
+        navigator.push(options);
+    });
+
+    handleClose = () => {
+        const {actions, navigator, onClose} = this.props;
+        if (this.refs.view) {
+            this.refs.view.zoomOut().then(() => {
+                actions.selectPost('');
```

Comment:
Why are we changing the selected post here?

Reply:
I need to clear it, if not this gives an error with the PostTextBox in the center channel